src/frontend/src/view/promote/promote.tsx

```diff
@@ -73,18 +78,27 @@ function mapStateToProps(state: AppState, ownProps: RouteComponentProps<PromoteR
     const deployNextEnv = selectDeploy(state, routeParams.app, environment.promotesTo);
 
     return {
-        isLoading: selectIsLoadingRelease(state),
+        isLoading: selectIsLoadingRelease(state) || selectIsLoadingIssues(state),
         release: selectReleaseWithCommits(state, routeParams.app, routeParams.env),
         fromVersion: deployNextEnv ? deployNextEnv.version : '',
         toVersion: deployCurrentEnv ? deployCurrentEnv.version : ''
     };
 }
 
 function mapDispatchToProps(dispatch: Dispatch<Action>): PromoteDispatchProps {
+
+    function hentIssues(commits: Commit[]) {
+        const issues = commits.map(commitToIssues)
+            .reduce((agg, current) => agg.concat(current), [])
+            .map((issue) => issue.name);
+        dispatch(getIssues(issues));
+    }
+
     return {
-        getCommits: (app: string, fromVersion: string, toVersion: string) => {
+        doGetInfoForPromote: (app: string, fromVersion: string, toVersion: string) => {
             dispatch(clearCommits());
-            dispatch(getCommitsForApplication(app, fromVersion, toVersion));
+            dispatch(getCommitsForApplication(app, fromVersion, toVersion))
```

Comment:
jeg har prøvd å putte logikken for å hente riktig data ut i action-creators. Fint om du fortsette det patternet.

Reply:
Ok. Det skjer to kall her, først mot commit-api, også mot jira-api. De har hver sin duck. Tenker du at jeg lager en ny fil, f.eks `promote-duck` som eksponerer `getInfoForPromote`, som da igjen kaller getCommitsForApplication og så henter issues?

Reply:
Riktig hele poenget er å gjøre mapDispatchToProps enkel, og heller legge implementasjonsdetaljene inn i action-creators som henter ned nødvendig data. Slik som jeg har gjort for release-not-duck.

Reply:
https://github.com/navikt/fo-forenklet-deploy/pull/13/commits/65e04e615b8fabf605d1d6e365e3f0550b965d88